src/components/drawer/Drawer.js

```diff
@@ -1,16 +1,40 @@
-import React, { useCallback } from 'react';
+import React, { useCallback, useRef, useEffect } from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
+import { disableBodyScroll, enableBodyScroll, clearAllBodyScrollLocks } from 'body-scroll-lock';
+
 import { useNavigation } from '../../hooks';
 
 const drawerClassName = 'react-navigation_drawer';
 
-const Drawer = ({ placement, withOverlay, children, className, overlayClassName }) => {
+const Drawer = ({
+    placement,
+    withOverlay,
+    lockBodyScroll,
+    children,
+    className,
+    overlayClassName,
+}) => {
+    const drawerRef = useRef();
     const { drawer } = useNavigation();
 
     const drawerClassNames = classNames(drawerClassName, `${drawerClassName}-${placement}`, className);
     const overlayClassNames = classNames(`${drawerClassName}-overlay`, overlayClassName);
 
+    useEffect(() => {
+        if (!lockBodyScroll) { return; }
+
+        if (drawer.isOpen) {
+            disableBodyScroll(drawerRef.current);
+        } else {
+            enableBodyScroll(drawerRef.current);
+        }
+    }, [drawer.isOpen, lockBodyScroll]);
+
+    useEffect(() => () => {
+        clearAllBodyScrollLocks();
```

Comment:
What does this method do exactly? Is it possible to have multiple locks?

Reply:
This function is useful if we have called `disableBodyScroll` for multiple target elements. It will clear them all. I've added this function as a safety measure to make sure that all body scroll locks are cleared when the component unmounts. But it's maybe unnecessary for our use case. What do you think?

Reply:
I can't think of a use case right now where calling this method would be bad or produce side effects. Nevertheless, we can simply call `enableBodyScroll(drawerRef.current)` anyway, right?

Reply:
Yap, that's right. I've changed it to call `enableBodyScroll` instead of `clearAllBodyScrollLocks()`. 
Thanks 👍